### PR TITLE
inspire_record: clean data before creating a record

### LINKED ISSUE #3098: holdingpen: content changed in the detail page can lead to invalid data
The detail page in the holdingpen allows to edit some content, but no validation is performed. This leads to accept content that is not valid, making the current workflow fail.

## Expected Behavior
Where possible, some cleanup is performed before saving the invalid data.

## Current Behavior
Data is accepted as it is and record creation fails due to invalid content.

## Steps to Reproduce (for bugs)

1. Submit a new article
2. Go in the holdinpen, to the halted submission
3. Remove all items from "Subject area"
4. Accept the article

### PR DESCRIPTION
**Pull request by ammirate**
Signed-off-by: Antonio Cesarano <cesarano2607@gmail.com>

Closes #3098 . 

## Checklist:

- [x] I have all the information that I need (if not, move to `RFC` and look for it).
- [x] I linked the related issue(s) in the corresponding commit logs.
- [x] I wrote [good commit log messages](https://github.com/torvalds/subsurface-for-dirk/blob/5f15ad5a86ada3c5e574041a5f9d85235322dabb/README#L92-L119).
- [x] My code follows the code style of this project.
- [x] I've added any new docs if API/utils methods were added.
- [x] I have updated the existing documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by bavaria95** [comment]
ci test please

**Comment by michamos** [comment]
list deduplication should not be done blindly. It can happen that two authors are the same on a paper.

**Comment by ammirate** [comment]
@michamos you are right. So, what do you suggest? Just don't  remove duplicates? In the end, the main thing is to remove empty fields.

File: tests/integration/test_records.py [hunk 1]
```diff
@@ -967,3 +969,35 @@ def test_create_with_skip_files_param_overrides_records_skip_files_conf_and_does
...
+        ],
+        # record/1628455/export/xme -- with some modification
+    }
+    non_valid = False
+    try:
+        validate(record_json)
+    except ValidationError:
+        non_valid = True
+
+    assert non_valid
```
Comment (jacquerie) [inline tests/integration/test_records.py]: All these lines could have been:
```python
with pytest.raises(ValidationError):
    validate(record_json)
```

Comment (ammirate) [reply tests/integration/test_records.py]: doh, late :/

**Pull request merged**